python/res/job_queue/ext_job.py

```diff
@@ -76,7 +76,10 @@ def __init__(self, config_file, private, name = None , license_root_path = None
             raise IOError('No such config file "%s".' % config_file)
 
     def __repr__(self):
-        return self._create_repr('%s, config_file = %s' % (self.name() , self.get_config_file()))
+        if self._address():
+            return self._create_repr('%s, config_file = %s' % (self.name() , self.get_config_file()))
```

Comment:
Since you're already changing this, you could change to .format instead of %, but not very important.